src/libsystemd/sd-device/device-monitor.c

```diff
@@ -249,10 +249,7 @@ _public_ int sd_device_monitor_new(sd_device_monitor **ret) {
 _public_ int sd_device_monitor_stop(sd_device_monitor *m) {
         assert_return(m, -EINVAL);
 
-        m->event_source = sd_event_source_unref(m->event_source);
-        (void) device_monitor_disconnect(m);
-
-        return 0;
+        return sd_event_source_set_enabled(m->event_source, SD_EVENT_OFF);
```

Comment:
hmm, this would also mean that the events will still be queued, right? just never read then, until the socket buffer runs over?

maybe if you leave the fd around here unsubscribe from all mcast groups, and then flush anything queued via flush_fd()?